lib/src/pubspec.dart

```diff
@@ -77,75 +74,64 @@ class Pubspec extends PubspecBase {
   ///
   /// This will be null if this was created using [new Pubspec] or [new
   /// Pubspec.empty].
-  final SourceRegistry _sources;
+  final SourceRegistry? _sources;
 
   /// The location from which the pubspec was loaded.
   ///
   /// This can be null if the pubspec was created in-memory or if its location
   /// is unknown.
-  Uri get _location => fields.span.sourceUrl;
+  Uri? get _location => fields.span.sourceUrl;
 
   /// The additional packages this package depends on.
-  Map<String, PackageRange> get dependencies {
-    if (_dependencies != null) return _dependencies;
-    _dependencies =
-        _parseDependencies('dependencies', fields.nodes['dependencies']);
-    return _dependencies;
-  }
+  Map<String, PackageRange> get dependencies => _dependencies ??=
+      _parseDependencies('dependencies', fields.nodes['dependencies']);
 
-  Map<String, PackageRange> _dependencies;
+  Map<String, PackageRange>? _dependencies;
 
   /// The packages this package depends on when it is the root package.
-  Map<String, PackageRange> get devDependencies {
-    if (_devDependencies != null) return _devDependencies;
-    _devDependencies = _parseDependencies(
-        'dev_dependencies', fields.nodes['dev_dependencies']);
-    return _devDependencies;
-  }
+  Map<String, PackageRange> get devDependencies => _devDependencies ??=
+      _parseDependencies('dev_dependencies', fields.nodes['dev_dependencies']);
 
-  Map<String, PackageRange> _devDependencies;
+  Map<String, PackageRange>? _devDependencies;
 
   /// The dependency constraints that this package overrides when it is the
   /// root package.
   ///
   /// Dependencies here will replace any dependency on a package with the same
   /// name anywhere in the dependency graph.
-  Map<String, PackageRange> get dependencyOverrides {
-    if (_dependencyOverrides != null) return _dependencyOverrides;
-    _dependencyOverrides = _parseDependencies(
-        'dependency_overrides', fields.nodes['dependency_overrides']);
-    return _dependencyOverrides;
-  }
+  Map<String, PackageRange> get dependencyOverrides =>
+      _dependencyOverrides ??= _parseDependencies(
+          'dependency_overrides', fields.nodes['dependency_overrides']);
+
+  Map<String, PackageRange>? _dependencyOverrides;
 
-  Map<String, PackageRange> _dependencyOverrides;
+  Map<String, Feature> get features => _features ??= _computeFeatures();
```

Comment:
Note to self: We really should remove 'features' it is experimental code that was never done.